support/Pipelines/Bbh/ControlId.py

```diff
@@ -0,0 +1,173 @@
+
+    - Measure the difference between the horizon quantities and the desired
+      values.
+
+    Arguments:
+      id_input_file_path: Path to the input file of the first initial data run.
+      id_run_dir: Directory of the first initial data run. If not provided, the
+        directory of the input file is used.
+      residual_tolerance: Residual tolerance used for termination condition.
+        (Default: 1.e-6)
+      max_iterations: Maximum of iterations allowed. Note: each iteration is
+        very expensive as it needs to solve an entire initial data problem.
+        (Default: 30)
+    """
+
+    # Read input file
+    if id_run_dir is None:
+        id_run_dir = Path(id_input_file_path).resolve().parent
+    with open(id_input_file_path, "r") as open_input_file:
+        _, id_input_file = yaml.safe_load_all(open_input_file)
+    binary_data = id_input_file["Background"]["Binary"]
+    M_target_A = binary_data["ObjectRight"]["KerrSchild"]["Mass"]
+    M_target_B = binary_data["ObjectLeft"]["KerrSchild"]["Mass"]
+    chi_target_A = binary_data["ObjectRight"]["KerrSchild"]["Spin"]
+    chi_target_B = binary_data["ObjectLeft"]["KerrSchild"]["Spin"]
+    x_B, x_A = binary_data["XCoords"]
+    separation = x_A - x_B
+    orbital_angular_velocity = binary_data["AngularVelocity"]
+    radial_expansion_velocity = binary_data["Expansion"]
+
+    # File to write control diagnostic data
+    data_file = open(f"{id_run_dir}/ControlParamsData.txt", "w")
+
+    iteration = 0
+    control_run_dir = id_run_dir
+
+    # Function to be minimized
+    def Residual(
+        M_Kerr_A: float,
+        M_Kerr_B: float,
+        chi_Kerr_A_x: float,
+        chi_Kerr_A_y: float,
+        chi_Kerr_A_z: float,
+        chi_Kerr_B_x: float,
+        chi_Kerr_B_y: float,
+        chi_Kerr_B_z: float,
+    ):
+        nonlocal iteration
+        nonlocal control_run_dir
+
+        if iteration > 0:
+            print(
+                "\n"
+                "=========================================="
+                f" Control of BBH Parameters ({iteration}) "
+                "=========================================="
+            )
+            control_run_dir = f"{id_run_dir}/ControlParams{iteration:02}"
+
+            # Run ID and find horizons
+            generate_id(
+                mass_a=M_Kerr_A,
+                mass_b=M_Kerr_B,
+                dimensionless_spin_a=[chi_Kerr_A_x, chi_Kerr_A_y, chi_Kerr_A_z],
+                dimensionless_spin_b=[chi_Kerr_B_x, chi_Kerr_B_y, chi_Kerr_B_z],
+                separation=separation,
+                orbital_angular_velocity=orbital_angular_velocity,
+                radial_expansion_velocity=radial_expansion_velocity,
+                run_dir=control_run_dir,
+                scheduler=None,
+            )
+
+        # Get black hole physical parameters
+        horizons_file = h5py.File(f"{control_run_dir}/Horizons.h5", "r")
+        M_A = horizons_file["AhA.dat"][0, 4]  # ChristodoulouMass
```

Comment:
Use `spectre.Visualization.ReadH5.to_dataframe` to read the data as a Pandas DataFrame, then use the names like "ChristodoulouMass" instead of column indices. That will ensure this keeps working when we add more horizon output quantities. There's an example how to do this with our H5 Python bindings in `Inspiral.py`, but `to_dataframe` also works with `h5py`.